testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -267,6 +270,9 @@ protected void insertTableRows(String tableName, String floatToDoubleType)
                         "  DATE '2000-04-13', " +
                         "  'abc', " +
                         "  'abc', " +
+                        "  'Bigger Value', " +
```

Comment:
```
0: jdbc:hive2://localhost:10000/default> create table t1 (string_to_char string) partitioned by (id bigint) stored as orc;
0: jdbc:hive2://localhost:10000/default> insert into t1 values ('Hi',1), ('Hi     ', 2), ('Hello', 3);
0: jdbc:hive2://localhost:10000/default> alter table t1 change column string_to_char string_to_char char(3);
0: jdbc:hive2://localhost:10000/default> select '"' || string_to_char || '"' from t1;
+--------+
|  _c0   |
+--------+
| "Hi"   |
| "Hi"   |
| "Hel"  |
+--------+
(edited)
```
```
trino:default> select '"' || string_to_char || '"'  from hive.default.t1;
 _col0 
-------
 "Hel" 
 "Hi " 
 "Hi " 
(3 rows)
```

It seems that Trino behaves slightly differently than Hive.

Reply:
Relevant code in Hive

https://github.com/linkedin/linkedin-orc/blob/7b695e34e38f27dd17784341a67326ea0db96903/java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java#L149-L160

Feel free to add it to the PR description if you consider it noteworthy.

Reply:
Discussed via Slack and agreed that the difference is related to a peculiarity of the Hive client.